src/UI/NewsSourceModal.jsx

```diff
@@ -8,50 +8,50 @@ import {
   DialogTitle,
   Divider,
   Stack,
-} from "@mui/material";
-import { useState } from "react";
-import { useNavigate } from "react-router-dom";
+} from '@mui/material';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useLocalStorage } from '../service/useLocalStorage';
+
+const DEFAULT_SOURCES = {
+  'The New York Times': true,
+  'News Api': true,
+  ' Gnews': true,
+};
 
 const NewsSourceModal = ({ isOpen, toggleModal }) => {
   const navigate = useNavigate();
-  const [sources, setSources] = useState({
-    "The New York Times": true,
-    "News Api": true,
-    Gnews: true,
-  });
-  const [error, setError] = useState("");
 
-  const handleSourceChange = (source) => {
+  const [sources, setSources] = useLocalStorage('newsSources', DEFAULT_SOURCES);
+  const [error, setError] = useState('');
+
+  const handleSourceChange = source => {
     const updatedSources = {
       ...sources,
       [source]: !sources[source],
     };
     setSources(updatedSources);
 
-    // Clear the error message if at least one source is selected
-    const activeSources = Object.values(updatedSources).filter(
-      (value) => value
-    );
+    const activeSources = Object.values(updatedSources).filter(Boolean);
     if (activeSources.length > 0) {
-      setError("");
+      setError('');
     }
   };
 
-  const handleSave = (e) => {
+  const handleSave = e => {
     e.preventDefault();
-    const activeSources = Object.keys(sources).filter(
-      (source) => sources[source]
+    const activeSources = Object.keys(sources || {}).filter(
```

Comment:
Have you tried removing this fallback ` || {}` it should be good now with new changes in useLocalStorage hook